libs/trpc/src/lib/router/user/login.ts

```diff
@@ -0,0 +1,34 @@
+import { publicProcedure } from '../../trpc';
+import { z } from 'zod';
+import * as services from '@dnd-assistant/api-services';
+import { InvalidCredentialsError, UserNotFoundError } from '@dnd-assistant/api-services';
+import { TRPCError } from '@trpc/server';
+import { validateEmail, validatePassword } from '@dnd-assistant/shared-utils';
+
+export const login = publicProcedure
+.input(
+    z.object({
+      email: z.string().refine(validateEmail),
+      password: z.string().refine(validatePassword),
+    })
+  )
+.mutation(async ({ ctx, input }) => {
+  const { email, password } = input;
+  try {
+    const sessionToken = await services.login(email, password);
+    return sessionToken;
+  } catch (e) {
+    if (e instanceof UserNotFoundError) {
+      return new TRPCError({
+        message: 'User not found for the provided email',
+        code: "NOT_FOUND",
+      });
+    } if (e instanceof InvalidCredentialsError) {
+      return new TRPCError({
+        message: 'Password provided does not match what\'s stored',
```

Comment:
[non-blocking] Generally recommended security practice is to not tell the user whether it was an incorrect username or password, but rather that their credentials were incorrect. From a user's perspective it can be nicer to receive information on which was wrong, so I leave it up to you!

Reply:
Ah yeah forgot about that, since we aren't limiting number of password attempts before a forced reset, I'm going to go ahead and just send a generic message